src/types/utilities.ts

```diff
@@ -92,10 +92,11 @@ function isReadonlyPropertyFromMappedType(
 		!/^__@[^@]+$/.test(name as string)
 	)
 		return declaration.readonlyToken.kind !== ts.SyntaxKind.MinusToken;
-	return isPropertyReadonlyInType(
-		(type as unknown as { modifiersType: ts.Type }).modifiersType,
-		name,
-		typeChecker
+
+	const { modifiersType } = type as { modifiersType?: ts.Type };
```

Comment:
We might want to create a `hasModifiersType` utility function.
Lets do this in another PR though.